src/github.com/getlantern/errlog/errlog_test.go

```diff
@@ -53,22 +53,49 @@ func TestAnonymousError(t *testing.T) {
 	rr := &rawReporter{}
 	ReportTo(rr)
 	l := ErrorLoggerFor("my-proxy-package")
-	l.Log(errors.New("any error"))
-	expected := Error{
-		"my-proxy-package",
-		"*errors.errorString",
-		"any error",
-		"",
-		map[string]string{},
-		l.systemInfo,
-		nil,
-		nil,
-		nil,
+	err := errors.New("any error")
+	l.Log(err)
+	expected := &Error{
+		Source:     err,
+		GoPackage:  "my-proxy-package",
+		GoType:     "*errors.errorString",
+		Desc:       "any error",
+		Op:         "",
+		Extra:      map[string]string{},
+		UserLocale: userLocale,
 	}
-	assert.Equal(t, expected, *rr.err, "should log errors created by errors.New")
+	assert.Equal(t, expected, rr.err, "should log errors created by errors.New")
 
 	l.Log(fmt.Errorf("any error"))
-	assert.Equal(t, expected, *rr.err, "should log errors created by fmt.Errorf")
+	assert.Equal(t, expected, rr.err, "should log errors created by fmt.Errorf")
+}
+
+func TestStructuredError(t *testing.T) {
+	rr := &rawReporter{}
+	ReportTo(rr)
+	l := ErrorLoggerFor("my-proxy-package")
+	source := errors.New("any error")
+	err := &Error{
+		Source:    source,
+		GoPackage: "specified package",
+		GoType:    "specified type",
+		Desc:      "specified desc",
+		Op:        "specified op",
+		Extra: map[string]string{
+			"extra1": "specified extra1",
+		},
+	}
+	l.Log(err)
```

Comment:
Cool @oxtoacart ! Then we can rename it to `errors` package (which was the package name I initially use) and replace the standard one. I prefer `errors.New(string)` and `errors.Wrap(error)` instead of creating object directly. By changing the method signature to return `errors.Error`, we encourage caller to attach anything valuable to it with type casting.